packages/server/graphql/mutations/helpers/getCCFromCustomer.ts

```diff
@@ -1,24 +1,50 @@
 import Stripe from 'stripe'
 import {getStripeManager} from '../../../utils/stripe'
 
-export default async function getCCFromCustomer(customer: Stripe.Customer) {
-  if (typeof customer.default_source !== 'string') {
-    throw new Error(`Could not find default_source for ${customer.id}`)
+export default async function getCCFromCustomer(
+  customer: Stripe.Customer | Stripe.DeletedCustomer
+) {
+  if (customer.deleted) {
+    throw new Error(`Customer ${customer.id} has been deleted`)
   }
-  const manager = getStripeManager()
-  const card = await manager.retrieveSource(customer.id, customer.default_source)
-
-  if (!card) {
-    throw new Error(`Could not find a card for ${customer.id}`)
+  if (
+    typeof customer.default_source !== 'string' &&
```

Comment:
-1 Checking the types, it could be of type `Card` as well, which should suit us. Same below in other places where we only accept `string`